### PR TITLE
Extend pluginOptions to also cover gulp-sourcemaps

### PR DESCRIPTION
**Pull request by narthollis**
As per issue #64 

Currently this both the options for sourcemap.init and sourcemap.write compressed into the same option.

I can't see any option conflicts upstream, so I don't think this would be an issue.

### CODE REVIEW

**Comment by chmontgomery** [comment]
Thanks for the PR! Definitely on the right track. A couple things:

* There does seem to be one overlap in init/write options: debug. I'm not entirely sure if people use it for init and not write or visa versa, however to be more explicit about this I think we should support each individually, e.g.: 

```js
main: {
  scripts: [
    './*.js'
  ],
  styles: [
    './*.css'
  ],
  options: {
    pluginOptions: {
      'gulp-sourcemaps': {
      	init: {loadMaps: false, debug: true},
      	write: {addComment: false},
      	destPath: '../maps' // this is the string "maps" by default
      }
    }
  }
}
```

* Also, I could see people needing to supply different options to scripts vs styles. In order to support this we would structure it how we structure other options, but in this case use defaults in case you want to do the same for both streams, e.g.:

```js
main: {
  scripts: [
    './*.js'
  ],
  styles: [
    './*.css'
  ],
  options: {
    pluginOptions: {
      'gulp-sourcemaps': {
      	init: {debug: true},
      	write: {addComment: false},
      	destPath: '../maps',
      	scripts: {
			init: {loadMaps: false},
			write: {addComment: true} // overrides {addComment: false}
      	},
      	styles: {
			init: {loadMaps: true},
			destPath: 'maps' // overrides '../maps'
      	}
      }
    }
  }
}
```

Let me know if you'll be able to look into these items or have different ideas

**Comment by narthollis** [comment]
Yeah, this seams all quite reasonable.

I will get this sorted tonight and update the PR.

**Comment by narthollis** [comment]
I have updated this pull request with the suggested chnages.

File: lib/init-option-defaults.js [hunk 1]
```diff
@@ -11,5 +18,17 @@ module.exports = function (bundle) {
   bundle[BundleKeys.OPTIONS].pluginOptions['gulp-minify-css'] = bundle[BundleKeys.OPTIONS].pluginOptions['gulp-minify-css'] || {};
   bundle[BundleKeys.OPTIONS].pluginOptions['gulp-uglify'] = bundle[BundleKeys.OPTIONS].pluginOptions['gulp-uglify'] || {};
   bundle[BundleKeys.OPTIONS].pluginOptions['gulp-concat'] = bundle[BundleKeys.OPTIONS].pluginOptions['gulp-concat'] || {};
+
+  bundle[BundleKeys.OPTIONS].pluginOptions['gulp-sourcemaps'] = defaults(bundle[BundleKeys.OPTIONS].pluginOptions['gulp-sourcemaps'] || {}, sourcemapDefaults);
+
+  // This is to get a clone of the base options
+  var sourcemapBaseOpts = defaults({}, bundle[BundleKeys.OPTIONS].pluginOptions['gulp-sourcemaps']);
+  // Remove the style and script sub attributes
```
Comment (chmontgomery) [inline lib/init-option-defaults.js]: why is this necessary? `sourcemapDefaults` should never be modified (and no scripts/styles key exists in the obj above)?

File: lib/init-option-defaults.js [hunk 2]
```diff
@@ -11,5 +18,17 @@ module.exports = function (bundle) {
...
+  delete sourcemapDefaults[BundleKeys.SCRIPTS];
+  delete sourcemapDefaults[BundleKeys.STYLES];
```
Comment (chmontgomery) [inline lib/init-option-defaults.js]: I assume you mean `sourcemapBaseOpts` here instead of `sourcemapDefaults`?

Comment (narthollis) [inline lib/init-option-defaults.js]: Yeah, i thought i had fixed that typo sorry. I had fixed that one localy.

Comment (chmontgomery) [inline lib/init-option-defaults.js]: np, I just quick fixed it and merged with my changes

**Comment by chmontgomery** [comment]
this change has landed in `v2.23.0`. Thanks for contributing!

**Comment by chmontgomery** [comment]
please try it out and let me know if there are any issues
